versioned_hdf5/versions.py

```diff
@@ -98,6 +101,19 @@ def commit_version(version_group, datasets, *,
         create_virtual_dataset(f, version_name, name, slices, attrs=attrs,
                                fillvalue=fillvalue)
     version_group.attrs['committed'] = True
+    if timestamp is not None:
+        if isinstance(timestamp, datetime.datetime):
+            if timestamp.tzinfo != datetime.timezone.utc:
+                raise ValueError("timestamp must be in UTC")
+            version_group.attrs['timestamp'] = timestamp.strftime("%Y-%m-%d %H:%M:%S.%f%z")
+        elif isinstance(timestamp, np.datetime64):
+            version_group.attrs['timestamp'] = f"{timestamp.astype(datetime.datetime)}+0000"
+        else:
+            raise TypeError("timestamp data must be either a datetime.datetime or numpy.datetime64 object")
+    else:
+        ts = datetime.datetime.now(datetime.timezone.utc)
+        version_group.attrs['timestamp'] = ts.strftime("%Y-%m-%d %H:%M:%S.%f%z")
```

Comment:
Maybe the format string could be defined globally somewhere in its own variable?